test/library/packages/Sort/correctness/sortAllIdxTypes.chpl

```diff
@@ -19,13 +19,13 @@ proc testsize(param s) {
 
   var B : [1:uint(s)..254:uint(s)] int;
   forall i in B.domain do
-    B[i] = (B.size-i):int;
+    B[i] = (B.sizeAs(i.type)-i):int;
 
   test(B);
 
   var C : [-63:int(s)..63:int(s)] int;
   forall i in C.domain do
-    C[i] = C.size-i;
+    C[i] = C.sizeAs(i.type)-i;
```

Comment:
Noting that PR #17750 will also change this test. The change here in your PR still has signed integer overflow.

Reply:
Good point.  We could change this to `.sizeAs(int)` instead?  (I updated it in this way simply out of laziness—trying to preserve the previous behavior as much as possible without thinking about it much).

Reply:
Since there will be a conflict with PR #17750 either way I think you should just leave it as-is and try to merge this before my PR :)

Reply:
Oh, and maybe it wasn't obvious, but the change I made in that PR takes away the `.size` calls altogether.

Reply:
I admittedly haven't gotten my head around #17750 yet.  If we used `sizeAs(int)` here, would the integer overflow still occur?  If so, can you spell out for me a bit better what it is?  (My assumption from the reproducer on https://github.com/chapel-lang/chapel/issues/6160#issuecomment-842424446 was that the problem was that `.size` was returning `max(int(8))` before and then adding to / subtracting from it; but if it returned an `int(64)` — which is the same type stored in the array — wouldn't the overflow go away?

Reply:
Right, if we used a `size` returning an `int` then there wouldn't be overflow. The reason I adjusted the test in a different manner from that is that there might still be truncation when converting back to e.g. `int(8)` - meaning the test would still rely on 2's complement etc. when in reality I think we just wanted the indices in a non-sorted order. I though the test would be nicer if it was independent of such issues so I adjusted it to always use the reverse range iteration to get the starting array.

Reply:
> when converting back to e.g. int(8)

When would that conversion back occur?  For this statement, `int - int(8)` would use the `-(int, int)` case and then `C[i]` stores `int`.  But is there a conversion back later that I'm missing?

Reply:
Oh, you are right. I was imagining that we were storing to an `int(8)` element but we're not. However it's  still the case that the input array has the numbers in a harder to understand order based on 2's complement.

Reply:
> However it's arguably nicer for the test to have the values to sort in the original input range.

Oh, that's a good point.  So to make it safe in that regard (with a minimal modification to what's here), I think we'd need to do something like:

```chapel
    C[i] = (C.sizeAs(uint(s))-i:uint(s)):int(s);
```

does that seem right?  (rationale: `uint(s)` math is guaranteed to wrap around appropriately; then the cast to `int(s)` gets it back to the correct index set; then an implicit coercion to `int` takes place in the assignment.

Reply:
I think that would avoid the signed overflow issue, yes; at least with my understanding of #6160 which I still consider open.

Reply:
@mppf: I ended up making this change since (a) I feel responsible for having introduced a bug with this test and (b) it felt thematically related to the `.sizeAs()` routine (where I'd wrestled with phrasing this line a few different ways in this PR prior to the version you reviewed).  But I hadn't realized that you'd reviewed and approved this PR prior to doing so, so wanted to mention it here so that you weren't caught off-guard.

Reply:
For this Pr, I'm fine with any version that passes testing. Let's talk about what we want to end up with under #17750.